src/App.js

```diff
@@ -1,86 +1,215 @@
+import { getArticlesFromApi } from './utils/NewsApi';
+import { register, login, getUserData, saveArticle, deleteArticle, getArticlesFromDb } from './utils/MainApi';
+import { CurrentUserContext } from './contexts/CurrentUserContext';
 
 function App() {
+  const [currentUser, setCurrentUser] = useState({});
+  const [isLoggedIn, setLoggedIn] = useState(false);
   const [isModalOpen, setModalOpen] = useState(false);
-  const [isFormOpen, setFormOpen] = useState(false);
+  const [isLoginFormOpen, setLoginFormOpen] = useState(false);
   const [isPreloaderOpen, setPreloaderOpen] = useState(false);
-  const [isSuccessOpen, setSuccessOpen] = useState(false);
-  const [isLoggedIn, setLoggedIn] = useState(false);
-  const [isSignIn, setSignIn] = useState(true);
-  const [isNavOpen, setNavOpen] = useState(false);
+  const [isInfoOpen, setInfoOpen] = useState(false);
+  const [searchedArticles, setSearchedArticles] = useState([]);
+  const [savedArticles, setSavedArticles] = useState([]);
+
   const location = useLocation();
   const navigate = useNavigate();
 
+  // localStorage.clear()
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const user = await getUserData(localStorage.getItem('jwt').jwt);
+        setCurrentUser(user);
+        setLoggedIn(true);
+      } catch (error) {
+        return;
+      }
+
+      try {
+        const articlesFromDb = await getArticlesFromDb();
+        setSavedArticles(articlesFromDb);
+      } catch (error) {
+        setSavedArticles([]);
+      }
+      setSearchedArticles(JSON.parse(localStorage.getItem('lastSearch')) || []);
+    })();
+  }, [isLoggedIn]);
+
   useEffect(() => {
     const closeByEscape = (e) => {
       if (e.key === 'Escape') {
-        handleCloseModal();
+        closeAllPopups();
       }
     };
     document.addEventListener('keydown', closeByEscape);
     return () => document.removeEventListener('keydown', closeByEscape);
   }, [])
 
-  function handleOpenForm() {
-      setModalOpen(true);
-      setFormOpen(true);
+  function closeAllPopups() {
+    setInfoOpen(false);
+    setLoginFormOpen(false);
+    setModalOpen(false);
   }
 
-  function handleCloseModal() {
-    setModalOpen(false);
-    setFormOpen(false);
-    setSignIn(true);
+  function redirectToLogin() {
+    setLoginFormOpen(true);
+    setInfoOpen(false);
   }
 
   function handleLogOut() {
-    setLoggedIn(false);
     navigate('/');
+    localStorage.removeItem('jwt');
+    localStorage.removeItem('lastSearch');
+    setLoggedIn(false);
   }
 
-  function redirectToLogin() {
-    setSuccessOpen(false);
-    setFormOpen(true);
+  function handleLogIn() {
+    setModalOpen(true);
+    setLoginFormOpen(true);
+  }
+
+  async function handleLoginSubmit(event, email, password) {
+    try {
+      event.preventDefault();
+      const data = await login(email, password);
+      localStorage.setItem('jwt', data.token);
+      setLoggedIn(true);
+    }
+    catch (error) {
+      closeAllPopups();
```

Comment:
You need to close popups only in blocks `then` after a successful response from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well


Please, correct it everywhere